app/activity/[id]/components/activity-card.tsx

```diff
@@ -0,0 +1,82 @@
+'use client'
+
+import { Heading, Box, Flex, Spacer, Text, IconButton } from '@chakra-ui/react'
+import { FiClock, FiMapPin, FiLink2, FiEdit3, FiTrash2 } from 'react-icons/fi'
+import { Link } from '@/components/link'
+import { customColors } from '@/theme/color'
+import { ActivityCollectionQuery } from '@generated/api'
+
+type ActivityInfoProps = {
+  activityData: ActivityCollectionQuery | undefined
```

Comment:
I think you don't need to pass undefined in this component. This component does not care if the data is undefined, this component's job is showing data from parent.  So in page, it's better to pass data if it's not undefined otherwise you show the loading UI. ActivityInfo is same!